src/Servers/Kestrel/Core/src/Internal/Http/PathNormalizer.cs

```diff
@@ -240,5 +265,196 @@ public static unsafe bool ContainsDotSegments(byte* start, byte* end)
 
             return false;
         }
+
+        // In-place implementation of the algorithm from https://tools.ietf.org/html/rfc3986#section-5.2.4
+        public static unsafe int RemoveDotSegments(Span<char> input)
```

Comment:
Are these method worth looking at further optimizations (maybe not as part of this PR)?
I have the feeling that some tricks can be played here -- but didn't think enough about it so far.

Reply:
Note these are copies of the Span<byte> APIs above. We'll avoid optimizing them in the release branch, but you're welcome to take a closer look in main.

Reply:
> release branch

Ah, didn't realize it's not _main_. Thanks -- ignore the other comments then 😉.
Will have a look in main (a bit later).